src/tests/SFA.DAS.EmployerIncentives.Functions.PaymentsProcess.AcceptanceTests/Features/ValidatePayments.feature

```diff
@@ -8,15 +8,24 @@ Feature: ValidatePayments
 
 Scenario: When bank details have been validated successfully
 	Given a legal entity has pending payments with valid bank details
+	And the apprentice 'is in learning' is true 
 	When the payment process is run
 	Then successful validation results are recorded
 	And payment records are created
 	And pending payments are marked as paid
 	And future payments are not marked as paid
 
-Scenario: When bank details have failed validation
-	Given a legal entity has pending payments without bank details
+Scenario: When at least one validation check fails
+	Given a legal entity has pending payments with <BankDetailStatus> bank details
+	And the apprentice 'is in learning' is <IsInLearning> 
 	When the payment process is run
-	Then failed validation results are recorded
+	Then bank details validation check is <BankDetailsValidationResult>
+	Then apprentice is in learning check is <IsInLearningValidationResult>
 	And no payment records are created
 	And pending payments are not marked as paid
+
+Examples:
+	| BankDetailStatus | IsInLearning | BankDetailsValidationResult | IsInLearningValidationResult |
```

Comment:
Do we want to do this or just keep every scenario separate?  There will be quite a few more validation states so this would end up with a lot of permutations.  

Then have a separate test for all validations steps passed?

Reply:
I was thinking about this, and the tests examples can just allow every validation to pass except one. This will keep the number of permutations down to the minimum. The only permutation I can see is when we have a status of null or false (such as the IsInLearning flag).

Reply:
So in that case how about changing it to something like

Given a pending payment
When the "validationstep" validation fails
Then the "validationstep" validation result is recorded
And no payment records are created
And pending payments are not marked as paid

Examples

| validationstep | 
| BankDetailStatus |
| IsInLearning |
| HasLearning |
| ... |

Then each test is distinct and handle he data set up in the code rather than in the specification

Reply:
agreed